sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2288,45 +2290,56 @@ public <T> Mono<FeedResponse<T>> readMany(
                                     }
-
-                                    requestCharge += page.getRequestCharge();
-                                    // TODO: this does double serialization: FIXME
-                                    finalList.addAll(page.getResults().stream().map(document ->
-                                        ModelBridgeInternal.toObjectFromJsonSerializable(document, klass)).collect(Collectors.toList()));
-                                    aggregateRequestStatistics.addAll(BridgeInternal.getClientSideRequestStatisticsList(page.getCosmosDiagnostics()));
-                                }
-                                CosmosDiagnostics aggregatedDiagnostics = BridgeInternal.createCosmosDiagnostics(aggregatedQueryMetrics);
-                                BridgeInternal.addClientSideDiagnosticsToFeed(aggregatedDiagnostics, aggregateRequestStatistics);
-                                headers.put(HttpConstants.HttpHeaders.REQUEST_CHARGE, Double
-                                    .toString(requestCharge));
-                                FeedResponse<T> frp = BridgeInternal
-                                    .createFeedResponseWithQueryMetrics(
-                                        finalList,
-                                        headers,
-                                        aggregatedQueryMetrics,
-                                        null,
-                                        false,
-                                        false,
-                                        aggregatedDiagnostics);
-                                return frp;
-                            });
+                                    CosmosDiagnostics aggregatedDiagnostics = BridgeInternal.createCosmosDiagnostics(aggregatedQueryMetrics);
+                                    BridgeInternal.addClientSideDiagnosticsToFeed(aggregatedDiagnostics, aggregateRequestStatistics);
+                                    headers.put(HttpConstants.HttpHeaders.REQUEST_CHARGE, Double
+                                            .toString(requestCharge));
+                                    FeedResponse<T> frp = BridgeInternal
+                                            .createFeedResponseWithQueryMetrics(
+                                                    finalList,
+                                                    headers,
+                                                    aggregatedQueryMetrics,
+                                                    null,
+                                                    false,
+                                                    false,
+                                                    aggregatedDiagnostics);
+                                    return frp;
+                                })
+                                .doOnSubscribe(s -> startTime.set(Instant.now()))
```

Comment:
I think in case of multiple subscriptions, this will not work well. Because the same `startTime` reference is going to be shared between different subscriptions. 
So the calculation is going to be incorrect.